File: app/src/main/java/com/udacity/notepad/util/SpaceItemDecoration.java
package com.udacity.notepad.util;

import android.content.Context;
import android.graphics.Rect;
import android.view.View;

import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

public class SpaceItemDecoration extends RecyclerView.ItemDecoration {

    private final Context context;
    private final int dimenRes;
    private final int space;

    public SpaceItemDecoration(Context context, int dimenRes) {
        this.context = context;
        this.dimenRes = dimenRes;
        this.space = context.getResources().getDimensionPixelOffset(dimenRes);
    }

    private SpaceItemDecoration() {
        throw new RuntimeException();
    }

    @Override
    public void getItemOffsets(Rect outRect, View view, RecyclerView parent, RecyclerView.State state) {
        super.getItemOffsets(outRect, view, parent, state);
        int position = parent.getChildAdapterPosition(view);
        switch (getOrientation(parent)) {
            case LinearLayoutManager.VERTICAL:
                if (position != 0) outRect.top = space;
                break;
            case LinearLayoutManager.HORIZONTAL:
                if (position != 0) outRect.left = space;
        }
    }

    private int getOrientation(RecyclerView parent) {
        RecyclerView.LayoutManager lm = parent.getLayoutManager();
        if (lm instanceof LinearLayoutManager) {
            return ((LinearLayoutManager) lm).getOrientation();
        }
        return -1;
    }
}

File: app/src/main/java/com/udacity/notepad/util/SpaceItemDecoration.kt
package com.udacity.notepad.util

import android.content.Context
import android.graphics.Rect
import android.view.View
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView
import androidx.recyclerview.widget.RecyclerView.ItemDecoration

class SpaceItemDecoration(context: Context, dimenRes: Int) : ItemDecoration() {

    private val space: Int= context.resources.getDimensionPixelOffset(dimenRes)


    override fun getItemOffsets(outRect: Rect, view: View, parent: RecyclerView, state: RecyclerView.State) {
        super.getItemOffsets(outRect, view, parent, state)
        val position = parent.getChildAdapterPosition(view)
        when (getOrientation(parent)) {
            LinearLayoutManager.VERTICAL -> if (position != 0) outRect.top = space
            LinearLayoutManager.HORIZONTAL -> if (position != 0) outRect.left = space
        }
    }

    private fun getOrientation(parent: RecyclerView): Int {
        val lm = parent.layoutManager
        return (lm as? LinearLayoutManager)?.orientation?:-1
       /* return if (lm is LinearLayoutManager) {
            lm.orientation
        } else -1*/
    }
}